packages/perseus/src/widgets/expression/expression-validator.ts

```diff
@@ -0,0 +1,177 @@
+import * as KAS from "@khanacademy/kas";
+import {Errors} from "@khanacademy/perseus-core";
+import _ from "underscore";
+
+import {Log} from "../../logging/log";
+import KhanAnswerTypes from "../../util/answer-types";
+
+import getDecimalSeparator from "./get-decimal-separator";
+
+import type {Rubric, OnInputErrorFunctionType} from "./expression.d";
+import type {PerseusExpressionAnswerForm} from "../../perseus-types";
+import type {PerseusStrings} from "../../strings";
+import type {PerseusScore} from "../../types";
+import type {Score} from "../../util/answer-types";
+
+/* Content creators input a list of answers which are matched from top to
+ * bottom. The intent is that they can include spcific solutions which should
+ * be graded as correct or incorrect (or ungraded!) first, then get more
+ * general.
+ *
+ * We iterate through each answer, trying to match it with the user's input
+ * using the following angorithm:
+ * - Try to parse the user's input. If it doesn't parse then return "not
+ *   graded".
+ * - For each answer:
+ *   ~ Try to validate the user's input against the answer. The answer is
+ *     expected to parse.
+ *   ~ If the user's input validates (the validator judges it "correct"), we've
+ *     matched and can stop considering answers.
+ * - If there were no matches or the matching answer is considered "ungraded",
+ *   show the user an error. TODO(joel) - what error?
+ * - Otherwise, pass through the resulting points and message.
+ */
+export default function expressionValidator(
```

Comment:
I changed it. I personally prefer have the export next to the declaration though.

I wonder if for these kind of code style nits if we should either codify them (linting) or accept different options. Because when it's your preference vs my own preference, they're both valid preferences; but if it's the team/org preference vs my own preference, especially if it's something that could be linted and/or handled by a formatter, then we can remove the cognitive load of having to remember what specific people prefer.

Reply:
> I personally prefer have the export next to the declaration though.

Do you have this preference for default exports specifically? I think of the `default export` as special in that it's the "main" export of the file and so it's nice to have it in a consistent place in the file. I don't think this way for named exports.

Reply:
I like both `export function` and `export default function` because it signals that something is used externally and that signal is close to the declaration.